src/components/Tokens/TokenDetails/TokenDescription.test.tsx

```diff
@@ -0,0 +1,85 @@
+import { QueryResult } from '@apollo/client'
+import userEvent from '@testing-library/user-event'
+import { USDC_MAINNET } from 'constants/tokens'
+import { Chain, Exact, TokenProjectQuery, useTokenProjectQuery } from 'graphql/data/__generated__/types-and-hooks'
+import { useCurrency } from 'hooks/Tokens'
+import { mocked } from 'test-utils/mocked'
+import { validTokenProjectResponse, validUSDCCurrency } from 'test-utils/pools/fixtures'
+import { act, render, screen } from 'test-utils/render'
+
+import { TokenDescription } from './TokenDescription'
+
+jest.mock('graphql/data/__generated__/types-and-hooks', () => {
+  const originalModule = jest.requireActual('graphql/data/__generated__/types-and-hooks')
+  return {
+    ...originalModule,
+    useTokenProjectQuery: jest.fn(),
+  }
+})
+
+jest.mock('hooks/Tokens', () => {
+  const originalModule = jest.requireActual('hooks/Tokens')
+  return {
+    ...originalModule,
+    useCurrency: jest.fn(),
+  }
+})
+
+const tokenAddress = USDC_MAINNET.address
+
+describe('TokenDescription', () => {
+  beforeEach(() => {
+    mocked(useTokenProjectQuery).mockReturnValue(validTokenProjectResponse)
+    mocked(useCurrency).mockReturnValue(validUSDCCurrency)
+  })
+
+  it('renders token information correctly with defaults', () => {
+    const { asFragment } = render(<TokenDescription tokenAddress={tokenAddress} />)
+    expect(asFragment()).toMatchSnapshot()
+
+    expect(screen.getByText('USDC')).toBeVisible()
+    expect(screen.getByText('USDCoin')).toBeVisible()
+    expect(screen.getByText('Website')).toBeVisible()
+    expect(screen.getByText('Twitter')).toBeVisible()
+    expect(screen.getByText('Etherscan')).toBeVisible()
+    expect(screen.getByText('0xA0b8...eB48')).toBeVisible()
+  })
+
+  it('truncates description and shows more', async () => {
+    const shortDescription = 'USDC is a fully collateralized US dollar stablecoin. USDC is the bridge...'
+    const longDescription =
+      'USDC is a fully collateralized US dollar stablecoin. USDC is the bridge between dollars and trading on cryptocurrency exchanges.'
+    render(<TokenDescription tokenAddress={tokenAddress} />)
+    const descriptionContainer = screen.getByText(shortDescription)
+
+    expect(descriptionContainer).toHaveTextContent(shortDescription)
+    expect(descriptionContainer).not.toHaveTextContent(longDescription)
+
+    await act(() => userEvent.click(screen.getByText('Show more')))
+    expect(descriptionContainer).toHaveTextContent(longDescription)
+    expect(descriptionContainer).not.toHaveTextContent(shortDescription)
+    expect(screen.getByText('Hide')).toBeVisible()
+  })
+
+  it('copy address button hidden when flagged', async () => {
+    const { asFragment } = render(<TokenDescription tokenAddress={tokenAddress} showCopy={false} />)
```

Comment:
I think it's more intuitive to not show the copy button by default, e.g.
```
<TokenDescription tokenAddress={tokenAddress} /> // No copy button shown
<TokenDescription tokenAddress={tokenAddress} showCopy /> // Copy button is shown
```
Slightly more succinct since JSX implicitly treats the `showCopy` prop as being `true`.